### PR TITLE
Add time range to donations query

### PR DESCRIPTION
**Pull request by mohammadranjbarz**
related to #296 

@CarlosQ96  We get timeout when givback-calculation script wants to get donations , and donations numbers are huge ( now it's over 8,000) so I added time filter then I can just get donations related to a specific round

### CODE REVIEW

**Comment by mohammadranjbarz** [comment]
After merging this we can call this query with this data
```
{
  donations(
      fromDate:"20220210 00:00:00", 
      toDate:"20220210 19:39:02"
    ) {
    valueUsd  
    toWalletAddress
    createdAt
    # fromWalletAddress
    # toWalletAddress
    # transactionId
    # amount
    # status
    # transactionNetworkId
    # speedup
    # currency
    # isFiat
    isProjectVerified
    project {
      verified  
      giveBacks
      id
      walletAddress
    }
    user {
      id  
      walletAddress
      email
      name
    }
  }
}
```

The query supposed to be backward-compatible, and alongside implementing this I added `joi` package for validating input data, we can use it in other query/mutations

File: src/utils/validators/graphqlQueryValidators.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+const throwHttpErrorIfJoiValidatorFails = (
+  validationResult: ValidationResult,
+) => {
+  if (validationResult.error) {
+    throw new Error(validationResult.error.details[0].message);
+  }
+};
+
+export const getDonationsQueryValidator = Joi.object({
+  fromDate: Joi.string().pattern(filterDateRegex).messages({
```
Comment (CarlosQ96) [inline src/utils/validators/graphqlQueryValidators.ts:22]: super cool validation library.

**Review by CarlosQ96 — APPROVED** [review]
Great work Mohammad, looks good too me tests cases and modified endpoint backward compatible.

**Comment by mohammadranjbarz** [comment]
> Great work Mohammad, looks good too me tests cases and modified endpoint backward compatible.

Thanks For reviewing it

**Pull request merged**